### PR TITLE
[ArticleBundle] Abstract away duplicate code and extend

### PR DESCRIPTION
**Pull request by delboy1978uk**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | #1725

### CODE REVIEW

**Comment by delboy1978uk** [comment]
That's all the duplicate code warnings gone from Insights :-)

File: src/Kunstmaan/AdminListBundle/AdminList/FilterType/Traits/BooleanFilterTrait.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    /**
+     * @return string
+     */
+    public function getTemplate()
+    {
+        return 'KunstmaanAdminListBundle:FilterType:booleanFilter.html.twig';
+    }
+}
```
Comment (deZinc) [inline src/Kunstmaan/AdminListBundle/AdminList/FilterType/Traits/BooleanFilterTrait.php]: Can you add newlines to the end of your files, there are several once with this issue. :)

File: src/Kunstmaan/AdminListBundle/AdminList/FilterType/Traits/NumberFilterTrait.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: derekm
+ * Date: 12/1/17
+ * Time: 12:53 PM
+ */
+
```
Comment (deZinc) [inline src/Kunstmaan/AdminListBundle/AdminList/FilterType/Traits/NumberFilterTrait.php]: unnecessary docblock. ;)

File: src/Kunstmaan/AdminListBundle/Controller/AdminListController.php [hunk 3]
```diff
@@ -25,6 +25,8 @@
  */
 abstract class AdminListController extends Controller
 {
+    protected $tabPane;
```
Comment (deZinc) [inline src/Kunstmaan/AdminListBundle/Controller/AdminListController.php]: Docblock needed.

File: src/Kunstmaan/ArticleBundle/Controller/AbstractArticleAdminListController.php [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    {
+        $this->initAdminListConfigurator($request);
+        if (!isset($this->configurator)) {
+            $this->configurator = $this->createAdminListConfigurator();
+        }
+
+        return $this->configurator;
+    }
+
+    protected function initAdminListConfigurator(Request $request)
```
Comment (deZinc) [inline src/Kunstmaan/ArticleBundle/Controller/AbstractArticleAdminListController.php]: Docblock needed. ;)

File: src/Kunstmaan/ArticleBundle/Controller/AbstractArticleAdminListController.php [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    public function exportAction(Request $request, $_format)
+    {
+        return parent::doExportAction($this->getAdminListConfigurator($request), $_format, $request);
+    }
+
+    /**
+     * @return AdminListConfiguratorInterface
+     */
+    abstract public function createAdminListConfigurator();
+}
```
Comment (deZinc) [inline src/Kunstmaan/ArticleBundle/Controller/AbstractArticleAdminListController.php]: Fix the docblocks (request e.g.) and the newline at the end.

File: src/Kunstmaan/ArticleBundle/PagePartAdmin/Traits/ArticleConfiguratorTrait.php [hunk 6]
```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Kunstmaan\ArticleBundle\PagePartAdmin\Traits;
+
+trait ArticleConfiguratorTrait
```
Comment (deZinc) [inline src/Kunstmaan/ArticleBundle/PagePartAdmin/Traits/ArticleConfiguratorTrait.php]: docblock needed.

File: src/Kunstmaan/GeneratorBundle/Command/Traits/GenerationValidationTrait.php [hunk 7]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Kunstmaan\GeneratorBundle\Command\Traits;
+
+use InvalidArgumentException;
+
+trait GenerationValidationTrait
```
Comment (deZinc) [inline src/Kunstmaan/GeneratorBundle/Command/Traits/GenerationValidationTrait.php]: Docblock if you'd please. :)

File: src/Kunstmaan/NodeBundle/Event/CopyPageTranslationNodeEvent.php [hunk 8]
```diff
@@ -2,134 +2,8 @@
...
-use Kunstmaan\NodeBundle\Entity\HasNodeInterface;
-use Kunstmaan\NodeBundle\Entity\Node;
-use Kunstmaan\NodeBundle\Entity\NodeTranslation;
-use Kunstmaan\NodeBundle\Entity\NodeVersion;
-
-/**
- * CopyPageTranslationNodeEvent
- */
-class CopyPageTranslationNodeEvent extends NodeEvent
+class CopyPageTranslationNodeEvent extends PageTranslationNodeEvent
```
Comment (deZinc) [inline src/Kunstmaan/NodeBundle/Event/CopyPageTranslationNodeEvent.php]: You know what....

File: src/Kunstmaan/NodeBundle/Event/PageTranslationNodeEvent.php [hunk 9]
```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Kunstmaan\NodeBundle\Event;
+
+use Kunstmaan\NodeBundle\Entity\HasNodeInterface;
+use Kunstmaan\NodeBundle\Entity\Node;
+use Kunstmaan\NodeBundle\Entity\NodeTranslation;
+use Kunstmaan\NodeBundle\Entity\NodeVersion;
+
+class PageTranslationNodeEvent extends NodeEvent
```
Comment (deZinc) [inline src/Kunstmaan/NodeBundle/Event/PageTranslationNodeEvent.php]: Give me a docblock here. :)

File: src/Kunstmaan/NodeBundle/Event/RecopyPageTranslationNodeEvent.php [hunk 10]
```diff
@@ -2,134 +2,7 @@
...
-use Kunstmaan\NodeBundle\Entity\HasNodeInterface;
-use Kunstmaan\NodeBundle\Entity\Node;
-use Kunstmaan\NodeBundle\Entity\NodeTranslation;
-use Kunstmaan\NodeBundle\Entity\NodeVersion;
-
-/**
- * RecopyPageTranslationNodeEvent
- */
-class RecopyPageTranslationNodeEvent extends NodeEvent
+class RecopyPageTranslationNodeEvent extends PageTranslationNodeEvent
```
Comment (deZinc) [inline src/Kunstmaan/NodeBundle/Event/RecopyPageTranslationNodeEvent.php]: docblock

File: src/Kunstmaan/AdminListBundle/AdminList/FilterType/Traits/BooleanFilterTrait.php [hunk 11]
```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Kunstmaan\AdminListBundle\AdminList\FilterType\Traits;
+
+use Symfony\Component\HttpFoundation\Request;
+
+trait BooleanFilterTrait
```
Comment (deZinc) [inline src/Kunstmaan/AdminListBundle/AdminList/FilterType/Traits/BooleanFilterTrait.php]: Docblock here too.

File: src/Kunstmaan/AdminListBundle/AdminList/FilterType/Traits/DateFilterTrait.php [hunk 12]
```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Kunstmaan\AdminListBundle\AdminList\FilterType\Traits;
+
+use DateTime;
+use Symfony\Component\HttpFoundation\Request;
+
+trait DateFilterTrait
```
Comment (deZinc) [inline src/Kunstmaan/AdminListBundle/AdminList/FilterType/Traits/DateFilterTrait.php]: and docblock here

File: src/Kunstmaan/AdminListBundle/AdminList/FilterType/Traits/DateTimeFilterTrait.php [hunk 13]
```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Kunstmaan\AdminListBundle\AdminList\FilterType\Traits;
+
+use DateTime;
+use Symfony\Component\HttpFoundation\Request;
+
+trait DateTimeFilterTrait
```
Comment (deZinc) [inline src/Kunstmaan/AdminListBundle/AdminList/FilterType/Traits/DateTimeFilterTrait.php]: docblock

File: src/Kunstmaan/AdminListBundle/AdminList/FilterType/Traits/NumberFilterTrait.php [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+namespace Kunstmaan\AdminListBundle\AdminList\FilterType\Traits;
+
+
+use Symfony\Component\HttpFoundation\Request;
+
+trait NumberFilterTrait
```
Comment (deZinc) [inline src/Kunstmaan/AdminListBundle/AdminList/FilterType/Traits/NumberFilterTrait.php]: add a dockblock

File: src/Kunstmaan/AdminListBundle/AdminList/FilterType/Traits/StringFilterTrait.php [hunk 15]
```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Kunstmaan\AdminListBundle\AdminList\FilterType\Traits;
+
+use Symfony\Component\HttpFoundation\Request;
+
+trait StringFilterTrait
```
Comment (deZinc) [inline src/Kunstmaan/AdminListBundle/AdminList/FilterType/Traits/StringFilterTrait.php]: docblock

File: src/Kunstmaan/ArticleBundle/PagePartAdmin/Traits/ArticleConfiguratorTrait.php (hunk 6 above)
Comment (deZinc) [inline src/Kunstmaan/ArticleBundle/PagePartAdmin/Traits/ArticleConfiguratorTrait.php]: Can you arrange a docblock here?

**Comment by delboy1978uk** [comment]
done!

**Comment by sandergo90** [comment]
@deZinc Can you review again?

**Comment by sandergo90** [comment]
@deZinc can you review again ? 😉

**Comment by Numkil** [comment]
I think this is ok for merging when merge conflict is solved.

**Comment by delboy1978uk** [comment]
@Numkil I can't remember why all that code was removed on my branch, I think it was just to do with getting the class name? It might even have been yourself that said to take all of that out. Should I resolve it to the code on my side?

**Comment by Numkil** [comment]
you made this PR request to improve the insights score by removing excessive duplicate code. But the PR now has a merge conflict in the adminlist controller. Please Resolve that on your end.

**Comment by Devolicious** [comment]
Closing this due to inactivity

**Pull request closed without merging**